src/lexer.l

```diff
@@ -291,3 +298,15 @@ struct|union|enum       {
                                             std::string(yytext) + std::string("'")); }
 
 %%
+
+void yy_set_source_string(const std::string& s) {
+  source = s;
```

Comment:
Technically a copy of `source` isn't necessary, right? Can read directly from driver instance? All we needed was an indirection.